spring-data-mongodb/src/main/java/org/springframework/data/mongodb/repository/support/SpringDataMongodbSerializer.java

```diff
@@ -149,12 +150,15 @@ protected String asDBKey(@Nullable Operation<?> expr, int index) {
 
 	protected Object convert(@Nullable Path<?> path, @Nullable Constant<?> constant) {
 
+		MongoPersistentProperty property = getPropertyFor(path);
+
 		if (!isReference(path)) {
+			if(property != null && property.hasExplicitWriteTarget()) {
```

Comment:
I wonder why this is necessary as the Querydsl predicate is rendered into our `Query` object that is then converted using the domain type information.

Reply:
true, the query mapper will take care of the conversion. So it works fine on current _main_, let me get back to the reporter to see if that's an issue with earlier versions. I would have argued that `SpringDataMongodbSerializer` should already consider this based on its description, so it's worth updating that.